crypto/derive-secret/src/lib.rs

```diff
@@ -19,7 +19,51 @@ const RAW_BYTES: &[u8; 8] = b"rawbytes";
 #[derive(Debug, Copy, Clone, Encodable, Decodable)]
 pub struct ChildId(pub u64);
 
-/// Secret key that allows deriving child secret keys
+/// `DerivableSecret` represents a secret key that can be used to derive child
+/// secret keys. Fedimint on init generates a root key, and then derives child
+/// keys from it for each of the modules. Each module's child key can then be
+/// used to derive child keys for its specific use cases. This derivable secret
+/// pattern is especially useful for creating backups for both client and server
+/// side operations.
+///
+/// The `DerivableSecret` struct in this implementation is only used for
+/// deriving secret keys, not public keys. This allows us to support multiple
+/// crypto schemes for the different cryptographic operations used across the
```

Comment:
I'd probably add "in contrast to BIP32", otherwise it might be hard to understand because you can do `xPriv -- derive --> xPriv -- convert --> Public` and just not `xPriv -- convert --> xPub -- derive --> xPub/Public`.

Reply:
"Fedimint on init generates a root key, and then derives child keys from it for each of the modules. " is a very ambiguous sentence. Is this client or server? In this context I think we just care about what it does and not where / when other crates use it.